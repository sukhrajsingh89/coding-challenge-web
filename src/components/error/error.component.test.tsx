import React from "react";
import { render, cleanup, screen } from "@testing-library/react";

import ErrorComponent from "./error.component";

describe("<CatsList", () => {
  afterEach(() => {
    cleanup();
  });

  const renderComponent = () => {
    return render(<ErrorComponent error="Page not found" />);
  };

  it("it should render the component", () => {
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it("it should display error message", () => {
    const { asFragment } = renderComponent();
    const error = screen.getByTestId("error");
    expect(error).toBeInTheDocument();
  });
});
