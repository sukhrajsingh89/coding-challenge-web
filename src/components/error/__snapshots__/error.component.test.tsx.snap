// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatsList it should render the component 1`] = `
<DocumentFragment>
  <h3
    data-testid="error"
  >
    Page not found
  </h3>
</DocumentFragment>
`;
